### PR TITLE
rework wide mkdir/rmdir with correct EOL

### PR DESCRIPTION
**Pull request by mikerenfro**
Unintentionally closed https://github.com/DFHack/dfhack/pull/3465 when fixing my wrong EOL characters.

### CODE REVIEW

File: library/modules/Filesystem.cpp [hunk 1]
```diff
@@ -99,7 +99,13 @@ bool Filesystem::mkdir (std::string path)
 {
     int fail;
 #ifdef _WIN32
-    fail = ::_mkdir(path.c_str());
+    ::std::wstring wide_path;
+    int convertResult = MultiByteToWideChar(437, 0, path.c_str(), (int)strlen(path.c_str()), NULL, 0);
+    if (convertResult > 0) {
+        wide_path.resize(convertResult+10);
```
Comment (lethosor) [inline library/modules/Filesystem.cpp:105]: Why not `+1`?

File: library/modules/Filesystem.cpp [hunk 2]
```diff
@@ -99,7 +99,13 @@ bool Filesystem::mkdir (std::string path)
 {
     int fail;
 #ifdef _WIN32
-    fail = ::_mkdir(path.c_str());
+    ::std::wstring wide_path;
+    int convertResult = MultiByteToWideChar(437, 0, path.c_str(), (int)strlen(path.c_str()), NULL, 0);
```
Comment (lethosor) [inline library/modules/Filesystem.cpp:103]: Does this assume the input is CP437? If so, I don't think that's what we want, and I believe that came up in #3465 too. Requiring CP437 input would severely restrict the characters that callers of these functions can use in paths.

File: library/modules/Filesystem.cpp (hunk 1 above)
Comment (mikerenfro) [reply library/modules/Filesystem.cpp:105]: Mostly due to the source I was copy/pasting from (and lack of sleep).

File: library/modules/Filesystem.cpp (hunk 2 above)
Comment (mikerenfro) [reply library/modules/Filesystem.cpp:103]: I may have misread. Thought I was converting *from* CP437 to CP1252, If I mixed those up, then my mistake. Is 437 not the default in DF?

Comment (lethosor) [reply library/modules/Filesystem.cpp:103]: CP437 is the default for text displayed on the DF screen. It's very much not the default for text in our code. We have a few CP437 <> UTF8 conversion functions, so I think UTF8 would be a reasonable assumption.

**Pull request closed without merging**